pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -372,6 +374,21 @@ public String getReplicatorPrefix() {
         String id = TopicName.get(base).getSchemaName();
         SchemaRegistryService schemaRegistryService = brokerService.pulsar().getSchemaRegistryService();
         return schemaRegistryService.getSchema(id)
+                .exceptionally(t -> {
+                    if (t.getCause() != null
```

Comment:
This condition looks a little bit hacky.
1) we are only checking `t.getCause()`, IMHO we should have some utility method that traverses the full exception chain, looking for a SchemaException
2) a "non recoverable" SchemaException is not enough to say that this is a "Schema does not exist" exception, can we introduce some specific SchemaException subclass or method like `isRecoverable()` ?

Reply:
Thank you for your reply! 

For question 1, the condition is inspired by `SchemaRegistryServiceImpl#trimDeletedSchemaAndGetList(java.lang.String)`, that is, those schemas with unrecoverable exceptions would be regarded as deleted schemas. Therefore here I continue to use the design concept.

For question 2, I agree that unrecoverable SchemaException is not equivant to "Schema does not exist". However in schema storage deletion scenario, an unrecoverable exception, which includes `NoSuchLedgerException` and `NoSuchEntryException`, is enough to say that the schema has already been deleted by others and we can quit the process.

Reply:
Thanks for your clarification 

What about creating an utility method and reduce code duplication?